src/test/java/org/prebid/server/bidder/applogy/ApplogyBidderTest.java

```diff
@@ -0,0 +1,293 @@
+
+    @Test
+    public void creationShouldFailOnInvalidEndpointUrl() {
+        assertThatIllegalArgumentException().isThrownBy(() -> new ApplogyBidder("invalid_url"));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldSkipEmptyTokens() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(
+                impBuilder -> impBuilder
+                        .ext(mapper.valueToTree(ExtPrebid.of(null,
+                                ExtImpApplogy.of(null)))));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = applogyBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1)
+                .containsOnly(BidderError.badInput(
+                        "token is empty"));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldSkipInvalidImpressionAndAddError() {
+        // given
+        ExtPrebid<?, ExtImpApplogy> ext = ExtPrebid.of(null, ExtImpApplogy.of("token1"));
+        Imp imp = givenImp(
+                impBuilder -> impBuilder
+                        .banner(null)
+                        .id("2")
+                        .ext(mapper.valueToTree(ext))
+                        .banner(Banner.builder().w(300).h(400).build())
+        );
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(asList(
+                        imp,
+                        givenImp(impBuilder -> impBuilder
+                                .banner(null)
+                                .id("2")
+                                .ext(mapper.valueToTree(ext))
+                                .audio(Audio.builder().build()))))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = applogyBidder.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1)
+                .containsOnly(BidderError.badInput(
+                        "Applogy only supports banner, video or native ads"));
+        assertThat(result.getValue()).hasSize(1)
+                .extracting(httpRequest -> mapper.readValue(httpRequest.getBody(), BidRequest.class))
+                .flatExtracting(BidRequest::getImp)
+                .containsOnly(imp);
+    }
+
+
+    @Test
+    public void makeBidsShouldReturnErrorWhenResponseBodyCouldNotBeParsed() {
+        // given
+        final HttpCall<BidRequest> httpCall = givenHttpCall(null, "invalid");
+
+        // when
+        final Result<List<BidderBid>> result = applogyBidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1);
+        assertThat(result.getErrors().get(0).getMessage()).startsWith("failed to decode json");
+        assertThat(result.getErrors().get(0).getType()).isEqualTo(BidderError.Type.bad_server_response);
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeBidsShouldReturnEmptyResultWhenResponseWithNoContent() {
```

Comment:
Move test case down to where `makeBids` are located in accordance to the method's flow.